libdnf/base/transaction.cpp

```diff
@@ -352,9 +370,49 @@ Transaction::TransactionRunResult Transaction::Impl::run(
             transaction_problems.emplace_back((*it).to_string());
         }
         return TransactionRunResult::ERROR_CHECK;
-    };
+    }
+
+    rpmtransFlags rpm_transaction_flags{RPMTRANS_FLAG_NONE};
+    for (const auto & tsflag : config.tsflags().get_value()) {
+        bool found = false;
+        for (const auto & [string_name, enum_item] : string_tsflag_map) {
+            if (tsflag == string_name) {
+                rpm_transaction_flags |= static_cast<rpmtransFlags>(enum_item);
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            // TODO(jrohel) Use a logger properly and change a way how to report to terminal
+            auto msg = fmt::format("Invalid tsflag: {}", tsflag);
+            std::cout << msg << std::endl;
```

Comment:
I'd really like we didn't add any more `std::cout`s and `std::cerr`s into libdnf. It messes out the outputs and can cause issues in tests that are very hard to workaround.

I think the approach is wrong here. How do we handle misconfigurations in general? Shouldn't we throw an exception? We only have two generic mechanisms to report errors: logging and exceptions.

Exceptions exit the program, so they are sure to notify the user and they seem to be the way to go here (the validation can potentially happen earlier).

Logging won't exit the program, but it's bound to go unnoticed.

Reply:
I was considering options. Initiating a transaction that ignored (only logged) an invalid flag can be dangerous. I changed the code. An exception is now thrown.
But the change is in the related PR: https://github.com/rpm-software-management/libdnf/pull/1499 . Please merge it first.